### DISCUSSION TITLE
Weird usage with H3 Inverter software 2.0

### DISCUSSION THREAD

**Discussion by petzi0815** [nathanmarlor/foxess_modbus#648] [Q&A]
Hi guys,

i just upgraded to H3 2.0 and can see a weird behaviour. It seems that the data coming from the inverter is somehow messed up now and the grid usage almonst equals the feedin sum. The values that come via modbus are wrong as i can see on the meter that shows a different usage.

anyone else has similar problems or a solution?
fun fact: the problem is not consisent through the day but i noticed it again in the afternoon hours today:

<img width="803" alt="image" src="https://github.com/user-attachments/assets/f4d40669-bff6-4ae1-b8e6-914771dd1bfe">

within the foxess cloud there are also really weid reportings / peaks that make no sense at all:

<img width="1100" alt="image" src="https://github.com/user-attachments/assets/413e6ac3-d722-48ed-8676-495bf2e190d7">

that's the weird grid consumption that has changed since the 14th. I did not have any real grid consumption since then

<img width="1061" alt="image" src="https://github.com/user-attachments/assets/ba842707-312c-4b24-8d54-9c72c4777c6a">

In addition i found some remarks in this forum mentioning a similar behaviour to modbus reporting wrong values with the new upgrade because of the new compensation setting:
https://www.photovoltaikforum.com/thread/178210-erfahrungen-mit-fox-ess/?postID=3850786

update (18.07)

this morning i have some graphs that could provide more insides possibly:
<img width="790" alt="image" src="https://github.com/user-attachments/assets/0107819d-4440-40b6-898b-b48f02919c81">

<img width="1093" alt="image" src="https://github.com/user-attachments/assets/c8dfa949-bc1f-4e4d-b548-9e8de786ec84">

There was a battery SOC available so no need to pull from the grid. During the night i can see everyhting works well and i have no grid usage. The inverter tries to compensate by providing a small feed-in durin the nighttime.

as soon as i have solar, the battery charges again but also the grid usage starts (there was no high load / SOC was still high enough).
i assume there is some sort of correlationg / wrong modbus reports when there is a feedin. It seems that feedin is calculated on top of the grid usage when there is solar production / the inverter is not in standby / drain brom battery mode.

update:
i had a call with foxess tech support and they confirmed that something with regards to modbus has changed in the 2.0 update. unfortunately there were not aware yet what has changed in detail but have reported the issue to development in china.

**Comment by halfprice11** [comment]
Hi,

I did the update for master version on my FOX inverter yesterday.
now I am on version 2.0.
and I love the setting, to reduce the basic grid consumption!!

But I do see same strange behaviour in my home automation energy dashboard, too.
it does not calculate the delta, but simply makes a sum of grid consumption,

BUT inverter does always goes with grid consumption VALUE XYZ and same time grid feedin same VALUE XYZ.
my power meter does stand still at this time.

In Energy dashboard I do have totals only


Strage and not as expacted.

Would use the delta only.
Which is represented by Grid consumption CT.
In energy dashboard I am not able to choose this unfortunately


Any help and news are apreciated

**Comment by halfprice11** [comment]
I do have an update on this.

today I tried to use the integration of
"TonyM1958/HA-FoxESS-Modbus"

And this integration did show correct values for "Feed-in" vs "consumption"
So it is NOT only a Fox development team issue :-)
While your integration did work until version 1.98, I agree, there has been some changes within the fox registers.

Regards

**Comment by FozzieUK** [comment]
This is a weird one, if i'm understanding all of this correctly - Fox introduced a grid_compensation feature for the H3 in V2.0 firmware (and also H3 Pro?).

The net result of this is that grid consumption modbus register is no longer reporting the correct value but only when there is solar generation in which case it reports the same value as the Feed-in taking place ?

I assume the stats you see on the Fox cloud are similarly messed up ?

TonyM's integration has a number of post modbus templates that do a balancing calculation on the modbus values - and from what you have said these are reporting ok ?

The Foxes-Modbus integration reports the value of the inverters modbus registers without any manipulation, to date this has been working ok, but it sounds as if (at least for the H3 on v2 firmware) this is not correct on a number of the reported sensors.

Can you confirm which sensor names you are looking at that are incorrect, firstly on this integration - is it the power sensors, total sensors (or both)

And could you confirm the equivalent sensor names on Tony's integration that are reporting correctly.

It does sound that unless Fox 'fix' the incorrect reporting of the H3 inverter registers, the only way to fix this will be to handle some of these calculations externally in templates.

**Comment by petzi0815** [comment]
Hey, it's mainly about those two entities that have apparently wrong values:
sensor.foxessinverter_grid_consumption_energy_today (Netzbezug)
sensor.foxessinverter_feed_in_energy_today (Netzeinspeisung)

 Both sensors are set on the energy dashboard and have weird values as mentioned above.
<img width="278" alt="image" src="https://github.com/user-attachments/assets/23fcddc6-3b3b-4a28-8e56-b33dc3676de6">

Fox hotline has confirmed to me that modbus values / registers have changed with the 2.0 firmware.
I can also share the thoughts that is somehow influenced with the solar generation. i can see high mysterious numbers when there is high solar combined with high feed to grid. That moment also the consume from grid goes up for no reason.

Unfortunately no H3 downgrade seems to be possible to a < 2.0 firmware so it would be highly appreciated if this could be fixed.
Anyone knows how to create a bug out of this discussion thread?

when i look at todays stats it somehow feels that the consume from grid now contains of the real consumption from grid (based on the inverter pulling from grid) PLUS the amout of solar energy that was fed to the grid today. So somehow those feed-in sums are topped on the actual grid consumption. it simply doesnt make sense why there is always a high grid consumption when there is a high feed in, can someone confirm that? 


<img width="879" alt="image" src="https://github.com/user-attachments/assets/aa8d258c-d969-4857-80ee-4169762f2cce">

<img width="991" alt="image" src="https://github.com/user-attachments/assets/e032c3d4-4884-4167-b324-10063545eff7">

very strage that yesterday when there also was a high feed in the problem does not seemed to occur.
<img width="914" alt="image" src="https://github.com/user-attachments/assets/d831b9e9-3ad0-4cc2-9e2b-4234b7898dce">

**Reply by FozzieUK** [reply]
It does sound very much like Fox have broken their internal calculation for it's daily totals (which is a bug) if you go to the inverter and on the front screen select History, Inverter Yield, you will be able to see the various totals the inverter maintains.

Can you confirm the totals you see there for Consume and Feedin are showing what the modbus sensors `sensor.foxessinverter_grid_consumption_energy_today` (Netzbezug) and 
`sensor.foxessinverter_feed_in_energy_today` (Netzeinspeisung) are showing.

**Reply by petzi0815** [reply]
Yep. I can confirm the values are the same on the history setting of the inverter.
When i read my main meter it only shows 0.6 kw or so that have been pulled from the grid today (so the result between consume from grid - feedin).

Any ideas how to change this while foxess is hopefully deploying a fix ? with the 2.0 version the statistics are getting messed up :-(

**Reply by FozzieUK** [reply]
If the inverters total is wrong as it's passed straight through we can't do anything within the integration, but thinking on my feet it should be possible to build something in yaml that will calculate the total consumption from the Grid_CT R/S/T, and pass that to a Riemann sum which will produce an increasing total the dashboard can use (temporarily).

It's obvious the grid consumption is broken, is the Feed in total still correct ?

**Reply by petzi0815** [reply]
Nope. The feed in is messed up as well. reads 4,5 on the inverter will the meter says 1.7

**Comment by FozzieUK** [comment]
ok, I think the thing to do is to build a couple of sensors that will track the consumption / feed-in power sensors - and monitor them to make sure you are getting accurate readings - i've put together some yaml code to add to your configuration.yaml if thats ok.

These sensors need adding to the `template:` section

```
template:
  - sensor:
      - name: "Grid Consumption fix"
        unit_of_measurement: kW
        state_class: measurement
        device_class: power
        state: >
            {% set gcR = states("sensor.foxessinverter_grid_consumption_R")|round(3,default=0) %}
            {% set gcS = states("sensor.foxessinverter_grid_consumption_S")|round(3,default=0) %}
            {% set gcT = states("sensor.foxessinverter_grid_consumption_T")|round(3,default=0) %}
	    {% set gc = (gcR + gcS + gcT)|round(3) %}
            {{ gc }}
      - name: "Feed-in fix"
        unit_of_measurement: kW
        state_class: measurement
        device_class: power
        state: >
            {% set fiR = states("sensor.foxessinverter_feed_in_R")|round(3,default=0) %}
            {% set fiS = states("sensor.foxessinverter_feed_in_S")|round(3,default=0) %}
            {% set fiT = states("sensor.foxessinverter_feed_in_T")|round(3,default=0) %}
            {% set fi = (fiR + fiS + fiT)|round(3) %}
            {{ fi }}
```

Can you double check the sensor names are correct for your system and are showing as available.



.
These Riemann sums need to be added to your other sums in the `sensor:` section 
```
sensor:
  - method: left
    name: grid consumption today sum
    platform: integration
    round: 3
    source: sensor.grid_consumption_fix
    unit_time: h
  - method: left
    name: feed-in today sum
    platform: integration
    round: 3
    source: sensor.feed_in_fix
    unit_time: h
```

And finally these need to be added to the `utility_meter:` section with any others you have

```
utility_meter:
  grid_consumption_today_fix:
    source: sensor.grid_consumption_today_sum
    cycle: daily
  feed_in_today_fix:
    source: sensor.feed_in_today_sum
    cycle: daily
```

This will then prove that the individual power sensors when converted into energy are reporting as you expect - the utility meters (sensor.grid_consumption_today_fix and sensor.feed_in_today_fix) will reset daily and should hopefully reflect your actual usage. 

You will have to restart your HA for these sensors to configure, and it'll take a few hours before the utility meters start to show data.

**Comment by halfprice11** [comment]
I highly appreciate, your response!
I am not the one, knowing how to deal with all of this, but will look forward for feedback from petzi,

BTW, if we change some of the logic of the sensor in my energy dashboard (choose new sonsors, or template ones), the full history will be gone, right?
this would be ugly :-)

**Comment by petzi0815** [comment]
Thanks for providing a possible solution. I share the same concerns as halfprice. if i set the sensor up like this and remove the other ones from the energy dashboard i would loose my whole history, right ? or is there any other way to keep / copy of the old history to the newly created entitites to preserve the history ? nevertheless i'll go ahead and add the new sensors so that i can confirm the values would help in the end

**Reply by FozzieUK** [reply]
For now keep the original sensors in the dashboard, this is just to prove we can actually get accurate totals by using the power sensors - if it does we can work out how to keep old and new working in the dashboard.

I would hope in the longer term as the inverter totals themselves are wrong Fox will correct it, it will be affecting all H3 users and cloud statistics as well.

**Reply by petzi0815** [reply]
Ok. I've went ahead and created the sensors. let's wait for the day tomorrow to pass and then compare the results in the evening. hopefully tomorrow it will be sunny here ;-)

**Reply by petzi0815** [reply]
Good morning. This night the inverter was using the SOC of battery, so the new fox feature tried to compensate b providing a feedin.
The actual meter shows: 0.9 kwh from the grid and a feedin of 0.5kw
I assume thats the new "feature" but tbh. it cant compensate in real time, so if it only compensates from "numbers" pov it doesnt make really sense as the feed-in is payed less then the cost from the consume from grid... but that's another story if the feature to have a constant feed in really makes sense....

with regards to the new meters. i've created a little dashboard to show side-bye-side what the integrations shows vs. the values provided by the fix:
<img width="628" alt="image" src="https://github.com/user-attachments/assets/bd7b580c-0afb-48a3-ad82-aa5ccabe5dea">

The actual meter of my house shows a feed in from 0.5 kw and a consumption of 0.9 kw. So the feed in value reported by the inverter / original integration is correct but the pull from grid value still isnt. i would assume there is another variable that might has to be taken in place

<img width="480" alt="image" src="https://github.com/user-attachments/assets/b3ea7d38-01e1-4ae7-84a8-3aaa418ce5e6">

**Reply by FozzieUK** [reply]
Thanks, sorry that looks like a school boy error on my part - the actual sum needs to take into account that one phase could be consuming whilst another is feeding in (and vice versa) my original template sensor doesn't do that which is why it will be over reading on both of the sums.

The actual sums should look like this to take that into account -

```
      - name: "Grid Consumption fix"
        unit_of_measurement: kW
        state_class: measurement
        device_class: power
        state: >
            {% set gcR = states("sensor.foxessinverter_grid_consumption_R") | round(3,default=0) %}
            {% set gcS = states("sensor.foxessinverter_grid_consumption_S") | round(3,default=0) %}
            {% set gcT = states("sensor.foxessinverter_grid_consumption_T") | round(3,default=0) %}
            {% set fiR = states("sensor.foxessinverter_feed_in_R") | round(3,default=0) %}
            {% set fiS = states("sensor.foxessinverter_feed_in_S") | round(3,default=0) %}
            {% set fiT = states("sensor.foxessinverter_feed_in_T") | round(3,default=0) %}
            
            {% set result = (fiR-gcR) + (fiS-gcS) + (fiT-gcT) %}
            {% set gcsum = 0 if result > 0.0 else (result * -1.0) | round(3) %}
            {{ gcsum | abs }}
```

```
      - name: "Feed-in fix"
        unit_of_measurement: kW
        state_class: measurement
        device_class: power
        state: >
        state: >
            {% set gcR = states("sensor.foxessinverter_grid_consumption_R")|round(3,default=0) %}
            {% set gcS = states("sensor.foxessinverter_grid_consumption_S")|round(3,default=0) %}
            {% set gcT = states("sensor.foxessinverter_grid_consumption_T")|round(3,default=0) %}
            {% set fiR = states("sensor.foxessinverter_feed_in_R")|round(3,default=0) %}
            {% set fiS = states("sensor.foxessinverter_feed_in_S")|round(3,default=0) %}
            {% set fiT = states("sensor.foxessinverter_feed_in_T")|round(3,default=0) %}
            {% set result = (fiR-gcR) + (fiS-gcS) + (fiT-gcT) %}
            {% set fisum = 0 if result < 0.0 else result | round(3) %}
            {{ fisum | abs }}
```

Could you update the template sensors and refresh 'Templates' in Developer Tools and lets see how that runs.

**Reply by petzi0815** [reply]
All right. I'll try that one. The issue seems to be caused by the manager firmware btw. meanwhile fox downgraded my master firmware back to 1.69 but manager is still on 1.79. I asked fox to downgrade manager to 1.71 as well .

**Reply by FozzieUK** [reply]
no problem, hopefully they will fix it in the future but it will be nice to prove that we can calculate the grid consumption and feed-in correctly 👍

**Reply by petzi0815** [reply]
I'll post another update tomorrow when the daily stats are reset to have a better comparison but just from the current values it looks quite good. Thats approx when i changed the templates to the new definition. as i have good solar today and draining from battery the values look quite promising

<img width="1132" alt="image" src="https://github.com/user-attachments/assets/f085c43f-2017-4a87-987e-38e64f2af168">

**Comment by halfprice11** [comment]
hi. you have been active, during my absense :-)
to get it right, you are checking for a basic mistake within your code to be fixed?
This has been caused (made visible) after Fox did change something in their inverters, but you did analyse, your grid csonsumption / feedin did do some misscalculation anyway?

so if we do update your "fix" it should work for all of the users (no matter, which fox version they are using)?

am I right?

**Reply by petzi0815** [reply]
Foxx has downgraded me this morning on the old version. Manager 1.78 seems to be okay I validate if the problem also occurs here but unfortunately i'll not be able to report anylonger if the bugfix really works with 1.79 :-(

**Comment by halfprice11** [comment]
If you would get me some help, where exactely to put the new code to, I will try to make further investigation.

So I will need to open my configuration.yaml, template.yaml and sensors.yaml and add some stuff, right?
not sure, which to put where - sorry, I am not too fimilar with coding, more configuring

**Reply by petzi0815** [reply]
You can work with Helpers to get the job done. No need to adjust the config files here. I also prefer to work with helpers to get everyhting more organized. Are you familiar how to setup them ?

**Reply by halfprice11** [reply]
not to much, some hints would be helpful to help me out :-)
for a simple example, once I added some helpers for a sum of some numbers.
But never found a proper solution to get the difference done

anyway, I will try

**Reply by petzi0815** [reply]
Okidoki.

1. Go here: https://[homeassistant]/config/helpers
2. Click on create helper

grid_consumption_fix
1. Choose Template
<img width="339" alt="image" src="https://github.com/user-attachments/assets/bd72367c-caa7-4376-9aaf-9138e69cf714">
<img width="337" alt="image" src="https://github.com/user-attachments/assets/4e0afd9f-5075-48b4-ae0f-87fa7deb845a">



```
{% set gcR = states("sensor.foxessinverter_grid_consumption_R") | round(3,default=0) %}
            {% set gcS = states("sensor.foxessinverter_grid_consumption_S") | round(3,default=0) %}
            {% set gcT = states("sensor.foxessinverter_grid_consumption_T") | round(3,default=0) %}
            {% set fiR = states("sensor.foxessinverter_feed_in_R") | round(3,default=0) %}
            {% set fiS = states("sensor.foxessinverter_feed_in_S") | round(3,default=0) %}
            {% set fiT = states("sensor.foxessinverter_feed_in_T") | round(3,default=0) %}
            
            {% set result = (fiR-gcR) + (fiS-gcS) + (fiT-gcT) %}
            {% set gcsum = 0 if result > 0.0 else (result * -1.0) | round(3) %}
            {{ gcsum | abs }}
```




2. feed-in-fix
(settings same as above, but different template)

```
{% set gcR = states("sensor.foxessinverter_grid_consumption_R")|round(3,default=0) %}
            {% set gcS = states("sensor.foxessinverter_grid_consumption_S")|round(3,default=0) %}
            {% set gcT = states("sensor.foxessinverter_grid_consumption_T")|round(3,default=0) %}
            {% set fiR = states("sensor.foxessinverter_feed_in_R")|round(3,default=0) %}
            {% set fiS = states("sensor.foxessinverter_feed_in_S")|round(3,default=0) %}
            {% set fiT = states("sensor.foxessinverter_feed_in_T")|round(3,default=0) %}
            {% set result = (fiR-gcR) + (fiS-gcS) + (fiT-gcT) %}
            {% set fisum = 0 if result < 0.0 else result | round(3) %}
            {{ fisum | abs }}
```



3. add new helper - integral sensor left: grid consumption today sum
<img width="341" alt="image" src="https://github.com/user-attachments/assets/0e4d68d6-288a-4106-99bd-b99e1b107521">
<img width="260" alt="image" src="https://github.com/user-attachments/assets/7a03bbd4-7503-446d-91e2-8f863c4cec98">



4. add new helper - integral sensor left: feed-in today sum
<img width="347" alt="image" src="https://github.com/user-attachments/assets/7c9dcea1-c210-437d-8ed6-85ff0b47992b">
<img width="247" alt="image" src="https://github.com/user-attachments/assets/51653bba-8bf7-4af3-bb99-5048b8da8afb">


5. add new helper- utility meter: grid_consumption_today_fix
<img width="341" alt="image" src="https://github.com/user-attachments/assets/7d088a37-5c91-46bc-8467-ad979e89650c">
<img width="365" alt="image" src="https://github.com/user-attachments/assets/0551b240-e4f2-4df2-a06a-9a2a5cb5ceed">

6. add new helper- utility meter: feed_in_today_fix
<img width="339" alt="image" src="https://github.com/user-attachments/assets/27af978c-7f0e-446b-aed0-85670779f537">
<img width="355" alt="image" src="https://github.com/user-attachments/assets/1268cc64-1f58-4dde-b6f5-3afcc0d472f0">

**Reply by halfprice11** [reply]
wow - thank you. will do as soon, as I find the time

**Reply by halfprice11** [reply]
ok. up and running - I will check for results in some hours and come back to you

**Reply by halfprice11** [reply]
as of now I would say, it is working :-)

**Reply by FozzieUK** [reply]
That’s great, it sounds like the grid power sensors can still be relied on, so it’s just the daily totals the inverter is getting wrong and hopefully Fox will fix.

Can you keep an eye on it for a few days and if it’s all good we can go from there 👍

**Reply by halfprice11** [reply]
yes, I will do.
but meanwhile, what ever Fox will do - is that kind of change in your code of your integration helpfull anyway?
the new calculation does work fine, for old and new versions of Fox inverter, as I do see.

If you would like to build a small update file (escpecially for me only), I would offer to test the update, if all does work well.
So by simply doing the update of your integration, updating the full stuff, but only the helpers.
BTW. at the moment, the daily sum / total does not reset at midnight.

Regards

**Reply by petzi0815** [reply]
Now after beeing downgraded again i can confirm that your sensors seem to have the nearly values as from the default integration.
Maybe even a little more precise for the feed-in values than the feed in reported by the integration parameter itself.

<img width="680" alt="image" src="https://github.com/user-attachments/assets/63767757-3e8c-475a-8d4f-b0bb94cc7cbb">

**Reply by halfprice11** [reply]
so this fix is working for old and new version - even better. Same on my side.
Thank you for confirmation.

**Reply by FozzieUK** [reply]
> @halfprice11 
> BTW. at the moment, the daily sum / total does not reset at midnight.

The daily sum is a total_increasing value and will never reset to 0, it's main purpose is to convert the instant power readings (kW) into a time base (kWh) - which is then used by the utility_meters `grid_consumption_today_fix` etc.. which do reset 'cycle' daily at midnight.

I hope that Fox ESS will resolve this fairly quickly as it is an obvious error, but for now you can use the 2 new utility meters in the Energy Dashboard if you want.

### **@halfprice11 To update your Energy dashboard but not lose your history** , 
**Follow these steps** (and always take a backup first).

Go to Settings, Dashboards and select 'Energy' - in the Electricity Grid section under "Grid Consumption" click **add Consumption** and select the sensor `grid_consumption_today_fix` , and then under "Return to grid" click **add Return** and select the sensor `feed_in_today_fix` 


 **DO NOT DELETE the existing entries**, leave them both as we want to retain the history - we are just adding 2 new sources.

Then we have to stop the dashboard doubling up the grid consumption and feed-in - you do this by disabling the 'total' entities provided by the Foxess-Modbus integration. Click on Settings, Devices & Services - find the Foxess-Modbus integration and click '1 Device' - find the 'grid consumption today' entry, click on it , then click on the 'wheel' in the top right corner
![1](https://github.com/user-attachments/assets/b53dbc09-6730-4f9a-ba0d-bace579efc31)

and move the 'Enabled' so the slider is off, then click 'update'

![1](https://github.com/user-attachments/assets/0cead9c8-b57a-43c7-98b2-c18abb950efe)

Then repeat for the 'Feed-in Today'

That will disable the entities from updating but keep their history and stop duplication in the dashboard.

If you go back to the Energy Dashboard editor it will warn that the entities are disabled but it will work fine, and it will only use the grid and feed-in utility_meters as the source for data going forward.

When Fox fix the problem with the Totals, all you have to do is re-enable the 2 Foxess-Modbus entities and it will start working again.

**Comment by FozzieUK** [comment]
Hi, i’m being told there is a new firmware version for the H3 that corrects the grid consumption / feed in reporting - the versions are Manager 1.80 and Master 2.01. 

Can anyone confirm they have these versions and the grid consumption and feed in statistics are now reporting correctly.

**Reply by abertram** [reply]
Hi,

I own a H3-10.0 and I am using the modbus integration since a few days. I had to ask Fox ESS to update the firmware as it was not working initially. Modbus communication didn't work. After the update it (the modbus communication) is working. But I am not sure about correct values. Grid consumption total and feed-in total are always 0 kWh.

Software versions:

![image](https://github.com/user-attachments/assets/8d0b3948-83a0-4d86-8f9b-59eb2c4ca1a1)

Energy dashboard:

![image](https://github.com/user-attachments/assets/18223b7d-f0c2-413b-86e6-b1a9ac7bea2f)

Best,
Alex

**Reply by FozzieUK** [reply]
@abertram The AIO H3 doesn't expose all of the registers that the H3 hybrid does - if you look further up this post, there are instructions about how to build total sensors within home assistant that do the same job.

**Reply by abertram** [reply]
Thank you @FozzieUK. You are reffering to https://github.com/nathanmarlor/foxess_modbus/discussions/648#discussioncomment-10146376, correct?

Alex

**Reply by FozzieUK** [reply]
Yes that should solve the problem for you as it generates the totals from the raw consumption and feed in sensors.

Note that your entity names are made up of sensor.{inverter name}_{sensor_name} so in the example above `"sensor.foxessinverter_grid_consumption_R"` the inverter has been given a name when setting up as 'foxessinverter' and the sensor name is `grid_consumption_R`.

If you did not give your inverter a name during setup (most people don't) the sensors will simply be named as `"sensor.grid_consumption_R"` before you create the template in home assistant go to 'Developer Tools' click on 'STATE' and search for all the entities that contain 'grid_consumption' and 'feed_in' to confirm the entity name and that they are available.

**Reply by fabian1512** [reply]
To update your firmware by yourself:
just register an alternative email-adress an select that you're installer.
searching for your inverter-ID brings everything up
😀
